src/Facebook/FacebookBatchResponse.php

```diff
@@ -98,7 +98,7 @@ public function setResponses(array $responses)
     public function addResponse($key, $response)
     {
         $originalRequestName = isset($this->batchRequest[$key]['name']) ? $this->batchRequest[$key]['name'] : $key;
-        $originalRequest = isset($this->batchRequest[$key]['request']) ? $this->batchRequest[$key]['request'] : null;
+        $originalRequest = isset($this->batchRequest[$key]['request']) ? $this->batchRequest[$key]['request'] : new FacebookRequest;
```

Comment:
Instead of a `FacebookRequest`, I'd use an `UnknownFacebookRequest` which extends `FacebookRequest`.
That way, we don't break BC, and the user can check with a simple `instanceof` if the original request is there or not.